ethereum/oracles/src/noir/circuit/abi.ts

```diff
@@ -0,0 +1,23 @@
+import { Abi } from '@noir-lang/noirc_abi';
+
+export class CircuitAbi {
+  constructor(public abi: Abi) {}
+
+  public public(): Abi {
```

Comment:
I would be afraid to use return as it is (I would expect it is) reserved word. We will run into troubles for that sooner or later. #reservedKeyword

I would use `publicParameters` instead.